### ISSUE TITLE
jsum.digest mutates array

### ISSUE THREAD

**Issue by kubakrzempek** [fraunhoferfokus/JSum#9]
Hey!

First of all, thank you for the lib, it's great!

I have found a potential issue with `digest` function which mutates the incoming param if the param is an array.
```js
> const objectSum = require('jsum')
> const someObject = {a: 1, b: 1}
> objectSum.digest(someObject, 'SHA256', 'base64') // 'Ta1RrEHrc4YvzjdfroW6E3Ef0Z8bJtjksfn6QFw9Wt8='
> someObject // { a: 1, b: 1 } -- not mutated, as expected.
> const someArray = [{a: 1}, {b: 1}]
> objectSum.digest(someArray, 'SHA256', 'base64') // 'mwUss0mtg+NHEZw2M0OFN7ppRT6Kna66Lr4N8gxfEHg='
> someArray // [ '{"a":1}', '{"b":1}' ] -- mutated, strings instead of objects.
> 
```

I'm not sure if that's intentional/known side effect or not, but I was very surprised seeing this, as I use the array in other places as well and I'd rather see it unchanged.

I can either work on the fix (if this is a bug), or add a note about it in the README, just let me know what's the expected course of action here.

**Comment by 7c** [comment]
yeah, also figured this out, i believe mutating is probably not intended

**Comment by yan-foto** [comment]
This is a critical issue and no, the code does not meant to act this way! As you might have found out, I'm really short on time and would appreciate any PR that would fix this.

**Comment by l-monninger** [comment]
@yan-foto It looks like you resolved this at HEAD, but have not updated the NPM package. Is that accurate?

**Comment by schester44** [comment]
yikes, any update on this? I can look into submitting a PR if needed.

**Comment by zaros** [comment]
bump

**Comment by yan-foto** [comment]
@kubakrzempek @l-monninger there has been a collision issue at HEAD but it seems that this problem was already solved. I fixed the hash collision (see #8) and published to npm. See if the problem persists. Thanks :)


> bump

@zaros 90's forums called, they want their "bump" back. Fork, fix, contribute.

**Comment by kubakrzempek** [comment]
> @kubakrzempek @l-monninger there has been a collision issue at HEAD but it seems that this problem was already solved. I fixed the hash collision (see #8) and published to npm. See if the problem persists. Thanks :)
> 
> > bump
> 
> @zaros 90's forums called, they want their "bump" back. Fork, fix, contribute.

@yan-foto I confirm the release fixed the issue, thanks!
